tests/irdl/test_declarative_assembly_format.py

```diff
@@ -2041,3 +2041,133 @@ class MixedOp(IRDLOperation):
+
+
+@pytest.mark.parametrize(
+    "program, output, generic",
+    [
+        (
+            "test.default",
+            "test.default",
+            '"test.default"() <{"prop" = false}> {"attr" = false} : () -> ()',
+        ),
+        (
+            "test.default prop false opt_prop true",
+            "test.default",
+            '"test.default"() <{"prop" = false, "opt_prop" = true}> {"attr" = false} : () -> ()',
+        ),
+        (
+            '"test.default"() <{"prop" = false, "opt_prop" = true}> {"attr" = false} : () -> ()',
+            "test.default",
+            '"test.default"() <{"prop" = false, "opt_prop" = true}> {"attr" = false} : () -> ()',
+        ),
+        (
+            '"test.default"() <{"prop" = false}> {"attr" = false} : () -> ()',
+            "test.default",
+            '"test.default"() <{"prop" = false}> {"attr" = false} : () -> ()',
+        ),
+        (
+            "test.default prop true opt_prop false",
+            "test.default prop 1 opt_prop 0",
+            '"test.default"() <{"prop" = true, "opt_prop" = false}> {"attr" = false} : () -> ()',
+        ),
+        (
+            "test.default attr false opt_attr true",
+            "test.default",
+            '"test.default"() <{"prop" = false}> {"attr" = false, "opt_attr" = true} : () -> ()',
+        ),
+        (
+            '"test.default"() <{"prop" = false}> {"attr" = false, "opt_attr" = true} : () -> ()',
+            "test.default",
+            '"test.default"() <{"prop" = false}> {"attr" = false, "opt_attr" = true} : () -> ()',
+        ),
+        (
+            "test.default attr true opt_attr false",
+            "test.default attr 1 opt_attr 0",
+            '"test.default"() <{"prop" = false}> {"attr" = true, "opt_attr" = false} : () -> ()',
+        ),
+    ],
+)
+def test_default_properties(program: str, output: str, generic: str):
+    ctx = MLContext()
+    ctx.load_op(DefaultOp)
+
+    parsed = Parser(ctx, program).parse_operation()
+    assert isinstance(parsed, DefaultOp)
+
+    stream = StringIO()
+    printer = Printer(stream=stream)
+    printer.print_op(parsed)
+
+    assert output == stream.getvalue()
+
+    stream = StringIO()
+    printer = Printer(stream=stream, print_generic_format=True)
+    printer.print_op(parsed)
+
+    assert generic == stream.getvalue()
+
+
+def test_default_accessors():
+    ctx = MLContext()
+    ctx.load_op(DefaultOp)
+
+    parsed = Parser(ctx, "test.default").parse_operation()
+
+    assert isinstance(parsed, DefaultOp)
+
+    assert parsed.prop.value.data == 0
+
+    assert parsed.properties.get("opt_prop") is None
+
+    assert parsed.opt_prop is not None
+
+    assert parsed.opt_prop.value.data == 1
+
+    assert parsed.attr.value.data == 0
+
+    assert parsed.attributes.get("opt_attr") is None
+
+    assert parsed.opt_attr is not None
+
+    assert parsed.opt_attr.value.data == 1
+
+
+def test_generic_accessors():
```

Comment:
I feel this test particularly but also the one above it should probably not live here, but am not sure exactly where they should go

Reply:
Moving it to `test_operation_definition.py`